### PR TITLE
Open service ui with modifier + mouse click

### PR DESCRIPTION
**Pull request by mlunoe**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

**Comment by MatApple** [comment]
Looks good to me.

File: src/js/components/ServiceList.js [hunk 1]
```diff
@@ -35,8 +36,19 @@ let ServiceList = React.createClass({
     return !_.isEqual(this.props, nextProps) || changedState;
   },
 
-  handleServiceClick: function (serviceName) {
-    this.context.router.transitionTo('dashboard-panel', {serviceName});
+  handleServiceClick: function (service, event) {
+    // Open service in new window/tab if service has a web URL
+    if (service.webui_url &&
+      (event.ctrlKey || event.shiftKey || event.metaKey)) {
```
Comment (rcorral) [inline src/js/components/ServiceList.js]: Modifier keys are cmd for mac and ctrl for windows/linux

File: src/js/components/ServiceList.js [hunk 2]
```diff
@@ -35,8 +36,19 @@ let ServiceList = React.createClass({
     return !_.isEqual(this.props, nextProps) || changedState;
   },
 
-  handleServiceClick: function (serviceName) {
-    this.context.router.transitionTo('dashboard-panel', {serviceName});
+  handleServiceClick: function (service, event) {
+    // Open service in new window/tab if service has a web URL
+    if (service.webui_url &&
```
Comment (jfurrow) [inline src/js/components/ServiceList.js]: The `Service` struct has a `#getWebURL` method.

Comment (kennyt) [inline src/js/components/ServiceList.js]: unfortunately, this might not be using the service struct.

but i wonder if it should be

Comment (mlunoe) [inline src/js/components/ServiceList.js]: It doesn't, and yes maybe it should. I didn't see it as part of this PR, though

File: src/js/components/ServiceList.js (hunk 1 above)
Comment (mlunoe) [inline src/js/components/ServiceList.js]: This handles if you open a new tab or a new window with a modifier key, i.e. shift, meta (command on mac) or ctrl (for Windows, will still be work as secondary click for Mac). Here is a reference: http://stackoverflow.com/a/20087506/1008519

File: src/js/components/ServiceList.js (hunk 2 above)
Comment (jfurrow) [inline src/js/components/ServiceList.js]: That makes sense!

**Pull request merged**